src/coreclr/src/vm/comdelegate.cpp

```diff
@@ -789,41 +789,31 @@ static PCODE GetVirtualCallStub(MethodDesc *method, TypeHandle scopeType)
     return pTargetCall;
 }
 
-FCIMPL5(FC_BOOL_RET, COMDelegate::BindToMethodName,
-                        Object *refThisUNSAFE,
-                        Object *targetUNSAFE,
-                        ReflectClassBaseObject *pMethodTypeUNSAFE,
-                        StringObject* methodNameUNSAFE,
-                        int flags)
+BOOL QCALLTYPE COMDelegate::BindToMethodName(
+                    QCall::ObjectHandleOnStack delegate,
+                    QCall::ObjectHandleOnStack target,
+                    QCall::TypeHandle methodTypeHandle,
+                    PCWSTR pwzMethodName,
+                    int flags)
 {
-    FCALL_CONTRACT;
-
-    struct _gc
-    {
-        DELEGATEREF refThis;
-        OBJECTREF target;
-        STRINGREF methodName;
-        REFLECTCLASSBASEREF refMethodType;
-    } gc;
+    QCALL_CONTRACT;
 
-    gc.refThis    = (DELEGATEREF) ObjectToOBJECTREF(refThisUNSAFE);
-    gc.target     = (OBJECTREF) targetUNSAFE;
-    gc.methodName = (STRINGREF) methodNameUNSAFE;
-    gc.refMethodType = (REFLECTCLASSBASEREF) ObjectToOBJECTREF(pMethodTypeUNSAFE);
+    MethodDesc *pMatchingMethod = NULL;
 
-    TypeHandle methodType = gc.refMethodType->GetType();
+    BEGIN_QCALL;
```

Comment:
You need to switch back to COOP mode before working with object references.

Reply:
Ah, looks like I moved it too late. Pulling it up.